server.go

```diff
@@ -1944,6 +1953,33 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 	return nil
 }
 
+// SetSendCompressor sets the compressor that will be used when sending
+// RPC payload back to the client. It may be called at most once, and must not
+// be called after any event that causes headers to be sent (see SetHeader for
+// a complete list). Provided compressor is used when below conditions are met:
+//
+//   - compressor is registered via encoding.RegisterCompressor
+//   - compressor name exists in the client advertised compressor names sent in
+//     grpc-accept-encoding metadata.
+//
+// The context provided must be the context passed to the server's handler.
+//
+// The error returned is compatible with the status package.  However, the
+// status code will often not match the RPC status as seen by the client
+// application, and therefore, should not be relied upon for this purpose.
+func SetSendCompressor(ctx context.Context, name string) error {
+	stream, ok := ServerTransportStreamFromContext(ctx).(*transport.Stream)
+	if !ok || stream == nil {
+		return status.Errorf(codes.Internal, "grpc: failed to fetch the stream from the context %v", ctx)
+	}
+
+	if err := grpcutil.ValidateSendCompressor(name, stream.ClientAdvertisedCompressors()); err != nil {
+		return status.Errorf(codes.Internal, "grpc: failed to set send compressor %v", err)
```

Comment:
Also, could we differentiate the two error cases with different error messages:
1. when the provided compressor is not registered
2. when the provided compressor is not supported by the client.